src/EA.Weee.RequestHandlers/Shared/GetObligationSummaryRequestHandler.cs

```diff
@@ -14,19 +15,33 @@ public class GetObligationSummaryRequestHandler : IRequestHandler<GetObligationS
         private readonly IWeeeAuthorization authorization;
         private readonly IMapper mapper;
         private readonly IEvidenceStoredProcedures evidenceStoredProcedures;
+        private readonly IOrganisationDataAccess organisationDataAccess;
 
         public GetObligationSummaryRequestHandler(IWeeeAuthorization authorization,
             IMapper mapper,
-            IEvidenceStoredProcedures evidenceStoredProcedures)
+            IEvidenceStoredProcedures evidenceStoredProcedures,
+            IOrganisationDataAccess organisationDataAccess)
         {
             this.authorization = authorization;
             this.mapper = mapper;
             this.evidenceStoredProcedures = evidenceStoredProcedures;
+            this.organisationDataAccess = organisationDataAccess;
         }
 
         public async Task<ObligationEvidenceSummaryData> HandleAsync(GetObligationSummaryRequest message)
         {
-            authorization.EnsureCanAccessInternalArea();
+            if (message.InternalAccess)
+            {
+                authorization.EnsureCanAccessInternalArea();
+            }
+            else
+            {
+                authorization.EnsureCanAccessExternalArea();
+
+                var organisation = await organisationDataAccess.GetById(message.OrganisationId);
+
+                authorization.EnsureOrganisationAccess(organisation.Id);
```

Comment:
Doesnt look like organisation is used here. Think you could probably just do authorization.EnsureOrganisationAccess(message.OrganisationId);

Reply:
removed